# Data visualistion et statistiques descriptives : formaliser des kpi

## expliquer ce qu'est un kpi

le kpi ou indiateur de performance cle permet de mesurer la performance d'un methode, c'est une valeur metrique que l'on peut comparer et donc comparer pour selectionner la methode la plus efficace pour par exemple augmenter le panier moyen d'un consommateur.

## importer pandas et numpy

In [2]:
import numpy as np
import pandas as pd

## importer le dataset eval.csv

In [12]:
df = pd.read_csv("~/python/eval_data.csv")

## afficher les 10 premiere lignes du dataset

In [13]:
df.head(10)

Unnamed: 0  order_id  quantity                              item_name  \
0           0         1         1           Chips and Fresh Tomato Salsa   
1           1         1         1                                   Izze   
2           2         1         1                       Nantucket Nectar   
3           3         1         1  Chips and Tomatillo-Green Chili Salsa   
4           4         2         2                           Chicken Bowl   
5           5         3         1                           Chicken Bowl   
6           6         3         1                          Side of Chips   
7           7         4         1                          Steak Burrito   
8           8         4         1                       Steak Soft Tacos   
9           9         5         1                          Steak Burrito   

                                  choice_description item_price  
0                                                NaN        NaN  
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN        NaN  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                NaN     $1.69   
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25

## afficher la dim du dataset

In [14]:
df.shape

(4622, 6)

4622 lignes et 6 colonnes

## afficher et decrire les colonnes du dataset

In [15]:
df.dtypes

Unnamed: 0             int64
order_id               int64
quantity               int64
item_name             object
choice_description    object
item_price            object
dtype: object

on constate que le dataset n'a pas ete nettoye


### netoyage

In [31]:
df.isna().sum()

Unnamed: 0               0
order_id                 0
quantity                 0
item_name                0
choice_description    1246
item_price             100
dtype: int64

on voit que certains prix ne sont pas indiques ce qui peut etre genant

In [34]:
df["item_price"] = df["item_price"].str.replace(',','.')

In [36]:
df["item_price"].str.get(0).value_counts()

$    4522
Name: item_price, dtype: int64

y a que des dollar on peut remplacer

In [38]:
df['item_price'] = df["item_price"].str.slice(1,None).astype('float')

maintenant on va corriger les vides

In [45]:
for nom_col in ['item_price','choice_description']:
    idx= df.columns.get_loc(nom_col)
    df.insert(idx+1,nom_col+'_nan', df[nom_col].isna().astype('int'))

In [46]:
df['item_price'].max()

44.25

In [47]:
df['item_price'].min()

1.09

In [48]:
df['item_price'].mean()

7.57655019902698

In [49]:
df['item_price'].median()

8.75

n/etant pas sur quoi mettre je vais juste mettre 0 pour le prix des valeurs manquantes

In [51]:
df['item_price'].fillna(0)

0        0.00
1        3.39
2        3.39
3        0.00
4       16.98
        ...  
4617    11.75
4618    11.75
4619    11.25
4620     8.75
4621     8.75
Name: item_price, Length: 4622, dtype: float64

## afficher les stat descriptives du dataset

In [43]:
df.describe()

Unnamed: 0     order_id     quantity   item_price
count  4622.000000  4622.000000  4622.000000  4522.000000
mean   2310.500000   927.254868     1.075725     7.576550
std    1334.400802   528.890796     0.410186     4.223898
min       0.000000     1.000000     1.000000     1.090000
25%    1155.250000   477.250000     1.000000     3.990000
50%    2310.500000   926.000000     1.000000     8.750000
75%    3465.750000  1393.000000     1.000000     9.390000
max    4621.000000  1834.000000    15.000000    44.250000

## quel est l'item le plus commande

In [55]:
df['item_name'].value_counts()

Chicken Bowl                             726
Chicken Burrito                          553
Chips and Guacamole                      479
Steak Burrito                            368
Canned Soft Drink                        301
Steak Bowl                               211
Chips                                    211
Bottled Water                            162
Chicken Soft Tacos                       115
Chicken Salad Bowl                       110
Chips and Fresh Tomato Salsa             110
Canned Soda                              104
Side of Chips                            101
Veggie Burrito                            95
Barbacoa Burrito                          91
Veggie Bowl                               85
Carnitas Bowl                             68
Barbacoa Bowl                             66
Carnitas Burrito                          59
Steak Soft Tacos                          55
6 Pack Soft Drink                         54
Chips and Tomatillo Red Chili Salsa       48
Chicken Cr

In [21]:
df['item_name'].mode()

0    Chicken Bowl
dtype: object

l'item le plus commande est 'Chicken Bowl'

## combien d'item on ete commander au total

In [56]:
df['quantity'].sum()

4972

il a 4972 items qui on ete commandes au total

## combien de chiffre d'affaire a realiser la societe

In [57]:
ca = df['item_price']*df['quantity']

In [58]:
ca.sum()

38998.020000000004

## quel est le revenu moyen par commande

In [66]:
moy = df['item_price'].groupby(df['order_id'])

In [68]:
moy.sum().mean()

18.681112322791716

c'est le revenu moyen par commande

etant limite par le temps je ne pas tout nettoye mais meme si les resultats sont inexacts les methodes de calculs sont valables